packages/type-utils/src/TypeOrValueSpecifier.ts

```diff
@@ -160,12 +160,19 @@ function typeDeclaredInPackage(
   // Handle scoped packages - if the name starts with @, remove it and replace / with __
   const typesPackageName = packageName.replace(/^@([^/]+)\//, '$1__');
 
-  const matcher = new RegExp(`${packageName}|${typesPackageName}`);
+  let matcher = new RegExp(`${packageName}|${typesPackageName}`);
+
+  if (packageName.startsWith('node:')) {
+    matcher = new RegExp(packageName.substring(5));
+  }
   return declarationFiles.some(declaration => {
-    const packageIdName = program.sourceFileToPackageName.get(declaration.path);
+    const packageIdName =
+      program.resolvedModules.has(declaration.path) ||
+      program.sourceFileToPackageName.has(declaration.path);
```

Comment:
I just want to add a small note:

In case of `@types/node`(the ts definitions for node), `sourceFileToPackageName` doesn't list the `.d.ts` files.

Example:
```ts
import url from "node:url";
```
`sourceFileToPackageName` won't list the file - `url.d.ts`

Reply:
Bump @arka1002 - I don't believe this logic should be changed in this PR. If there's an existing bug or usage blocker _specific to `no-floating-promises`_ then that'd make sense. But if it's some pre-existing issue, like `@types/node` not listing `.d.ts` files, then that's something separate. Especially if the proposed solution involves using a TypeScript internal like `resolvedModules` for the first time.